### PR TITLE
Calc changes additions/deletions per patches

### PR DESCRIPTION
**Pull request by La0**
Hello @marco-c & @calixteman,

I need to display the changes additions & deletions in Shipit (as +XX/-YY for example).
Whatthepatch does not give directly this information, so i had to compute it.

Regarding the unit test, i put all the patches checks in separate files, more easily diffable than a Python inline dict (see the second commit).

### CODE REVIEW

File: tests/test_patchanalysis.py [hunk 1]
```diff
@@ -36,12 +38,22 @@ def assertWarnings(self, warnings, expected_warnings):
         self.assertEqual(len(missed_warnings), 0)
         self.assertEqual(len(unexpected_warnings), 0)
 
-    def assertDictEqual(self, d1, d2, msg=None):
```
Comment (marco-c) [inline tests/test_patchanalysis.py]: Can you leave this override in place? It should give a more undestandable error when something fails.

**Comment by marco-c** [comment]
What is the plan for the dashboard? Do you want to show the changes per patch?
Shouldn't a we only care about the total size of the changes?

File: tests/test_patchanalysis.py (hunk 1 above)
Comment (La0) [inline tests/test_patchanalysis.py]: I can of course, it just didn't seem used anymore

File: tests/test_patchanalysis.py [hunk 2]
```diff
@@ -36,12 +38,22 @@ def assertWarnings(self, warnings, expected_warnings):
...
+            local_patches = json.load(f)
 
-        return True
+        # Compare str keys
+        keys = set(map(str, patches.keys()))
+        self.assertEqual(len(keys.symmetric_difference(list(local_patches.keys()))), 0)
+
+        # Keys can be int or str
+        for k, patch_data in patches.items():
+            self.assertDictEqual(patch_data, local_patches[str(k)])
```
Comment (La0) [inline tests/test_patchanalysis.py]: Ok, will do :+1:

**Comment by La0** [comment]
I _think_ the plan is to show these numbers per patche. I just asked Sylvestre for confirmation, but he preferred to see the details of additions/deletions instead of total changed lines.

The actual changes_size computation represents the total size of the patch, with the surrounding lines (non modified), it does not show the real number of modified lines (hence the difference between changes_size and changes_add + changes_del)

**Comment by marco-c** [comment]
> I think the plan is to show these numbers per patch.

This is actually unrelated to your PR, I'm not sure why I brought it up in the first place 😄

File: tests/test_patchanalysis.py (hunk 1 above)
Comment (marco-c) [inline tests/test_patchanalysis.py]: It's only useful to make the error message more precise when there's a failure.

**Comment by marco-c** [comment]
@La0 once you fix the minor issue, I will merge this, then I will release a new version after #56 is merged.

**Comment by La0** [comment]
I updated the PR to rebase and fix the assertDictEqual issue.

**Pull request merged**